packages/nx-plugin/src/generators/plugin/plugin.ts

```diff
@@ -89,55 +92,65 @@ function updatePluginConfig(host: Tree, options: NormalizedSchema) {
 
 export async function pluginGenerator(host: Tree, schema: Schema) {
   const options = normalizeOptions(host, schema);
+  const tasks: GeneratorCallback[] = [];
+
+  tasks.push(
+    await jsLibraryGenerator(host, {
+      ...schema,
+      config: 'project',
+      bundler: options.bundler,
+      publishable: true,
+      importPath: options.npmPackageName,
+      skipFormat: true,
+    })
+  );
 
-  await jsLibraryGenerator(host, {
-    ...schema,
-    config: 'project',
-    bundler: options.bundler,
-    importPath: options.npmPackageName,
-    skipFormat: true,
-  });
-
-  addDependenciesToPackageJson(
-    host,
-    {
-      '@nx/devkit': nxVersion,
-      tslib: tsLibVersion,
-    },
-    {
-      '@nx/jest': nxVersion,
-      '@nx/js': nxVersion,
-      '@nx/nx-plugin': nxVersion,
-      '@swc-node/register': swcNodeVersion,
-    }
+  tasks.push(
+    addDependenciesToPackageJson(
+      host,
+      {
+        '@nx/devkit': nxVersion,
+        tslib: tsLibVersion,
+      },
+      {
+        '@nx/jest': nxVersion,
+        '@nx/js': nxVersion,
+        '@nx/nx-plugin': nxVersion,
+        '@swc-node/register': swcNodeVersion,
```

Comment:
Use `{ addSwcRegisterDependencies } from '@nx/js/src/utils/add-swc-dependencies';`

Reply:
rebased and this is already implemented